src/main/java/org/opentripplanner/graph_builder/module/osm/OpenStreetMapModule.java

```diff
@@ -514,80 +517,55 @@ private void processParkAndRideNodes(Collection<OSMNode> nodes, boolean isCarPar
         new VehicleParkingEdge(parkVertex);
       }
 
+      if (!vehicleParkingToAdd.isEmpty()) {
+        vehicleParkingService.updateVehicleParking(vehicleParkingToAdd, List.of());
+      }
+
       LOG.info("Created {} {} P+R nodes.", n, isCarParkAndRide ? "car" : "bike");
     }
 
     private void buildBikeParkAndRideAreas() {
       LOG.info("Building bike P+R areas");
       List<AreaGroup> areaGroups = groupAreas(osmdb.getBikeParkingAreas());
-      int n = 0;
-      for (AreaGroup group : areaGroups) {
-        if (buildParkAndRideAreasForGroup(group, false)) n++;
-      }
-      if (n > 0) {
+      List<VehicleParking> vehicleParkingToAdd = buildParkAndRideAreasForGroups(areaGroups, false);
+      if (!vehicleParkingToAdd.isEmpty()) {
         graph.hasBikeRide = true;
+        VehicleParkingService vehicleParkingService = graph.getVehicleParkingService();
+        vehicleParkingService.updateVehicleParking(vehicleParkingToAdd, List.of());
       }
-      LOG.info("Created {} bike P+R areas.", n);
-    }
-
-    private void buildWalkableAreas(boolean skipVisibility, boolean platformEntriesLinking) {
```

Comment:
Was it intentional to change this file so heavily? Was it prettier reordering the methods?

Reply:
I got annoyed while browsing the code because there was a random method in between like 8 park and ride related methods so I moved it elsewhere as there was no logical reason for it to be there. However, I probably should have done it in an own commit because the diff is a bit messy.

This class is pretty big, I think we should look into if we could, for example, move some of the park and ride code elsewhere.

Reply:
Yes, this class is a mess and needs an urgent refactoring!